File: data-plane/simulators/gps_sim.go
package simulators

import (
	"bytes"
	"encoding/json"
	"fmt"
	"math/rand"
	"net/http"
	"time"

	"measure-it.com/central-server/models"
)

type GPSSimConfig struct {
	Min         int
	Max         int
	FrequenceMS int
	RandomNoise int
}

func GPSSimulator(name string, stop chan int8, update chan GPSSimConfig) {
	simConfig := GPSSimConfig{
		Min:         60,
		Max:         180,
		FrequenceMS: 2000,
		RandomNoise: 0,
	}

	gpsData := models.Dataframe{
		SensorName: name,
	}

	for {
		select {
		case simConfig = <-update:
			fmt.Println("Update ->", simConfig)
		case <-stop:
			return
		default:
			gpsData.SpeedKPH = rand.Int63() * 100
			jsonData, err := json.Marshal(gpsData)
			if err != nil {
				fmt.Println("Error Marshalling data:", err)
				return
			}

			url := "http://localhost:8080/sensorData"
			resp, err := http.Post(url, "application/json", bytes.NewBuffer(jsonData))
			if err != nil {
				fmt.Println("Error sending request:", err)
				return
			}

			if resp.StatusCode != 200 {
				fmt.Println("Failed request, status code:", resp.StatusCode)
				return
			}

			defer resp.Body.Close()
			time.Sleep(time.Millisecond * time.Duration(simConfig.FrequenceMS))
		}
	}
}

File: ss/gps/simulator/gps_sim.py
import json
import requests
import time

f = open("trackday_session_2.json", "r")
data = json.load(f)
# longitude_min, longitude_max = 1000, 0
# latitude_min, latitude_max = 1000, 0
f.close()

for d in data:
    if 'latitude' in d:
        requests.post("http://localhost:8080/sensorData", json={"sensorName": "gps-position-1", "longitude": d["longitude"], "latitude": d["latitude"]})
        time.sleep(0.1)
#     if d["x"] < longitude_min:
#         longitude_min = d['x']
#     if d["x"] > longitude_max:
#         longitude_max = d['x']

#     if d["y"] < latitude_min:
#         latitude_min = d['y']
#     if d["y"] > latitude_max:
#         latitude_max = d['y']

# print(longitude_min, longitude_max)
# print(latitude_min, latitude_max)